app/controllers/article_controller.rb

```diff
@@ -284,82 +217,38 @@ def record_deed
-    from_article.title = 'TO_BE_DELETED:'+old_from_title
-    from_article.save!
-
-    # walk through all pages referring to from_article
-    # walk through all pages referring to this
-    for link in from_article.page_article_links
-      source_text = link.page.source_text
-      link.page.rename_article_links(old_from_title, to_article.title)
-      link.page.save!
-      logger.debug("DEBUG: changed \n#{source_text} \nto \n#{link.page.source_text}\n")
-    end
-    # walk through all articles referring to this
-    for link in from_article.target_article_links
-      source_text = link.source_article.source_text
-      link.source_article.rename_article_links(old_from_title, to_article.title)
-      link.source_article.save!
-      logger.debug("DEBUG: changed \n#{source_text} \nto \n#{link.source_article.source_text}\n")
-    end
-
-    for link in from_article.source_article_links
-      link.destroy
-    end
-    # thankfully, rename_article_links is source-agnostic!
-
-    # change links
-    Deed.where(:article_id => from_article.id).update_all("article_id='#{to_article.id}'")
+  private
 
-    # append old from_article text to to_article text
-    if from_article.source_text
-      if to_article.source_text
-        to_article.source_text += from_article.source_text
-      else
-        to_article.source_text = from_article.source_text
-      end
-    end
-    # write to the DB
-    to_article.save!
-    from_article.destroy
+  def authorized?
+    redirect_to dashboard_path unless user_signed_in?
   end
 
-  def gis_truncated?(params, dec)
-    unless params[:latitude] || params[:longitude] then return false end
-
-    lat = params[:latitude].split('.')
-    lon = params[:longitude].split('.')
-
-    if lat.length == 2 then lat_dec = lat.last.length else lat_dec = 0 end
-    if lon.length == 2 then lon_dec = lon.last.length else lon_dec = 0 end
-
-    if lat_dec > dec || lon_dec > dec then return true else return false end
+  def article_params
+    params.require(:article).permit(:title, :uri, :source_text, :latitude, :longitude)
   end
 
   def sort_vertically(articles)
-    rows = (articles.length().to_f / LIST_NUM_COLUMNS).ceil
-    vertical_articles = Array.new(rows) { Array.new(LIST_NUM_COLUMNS) } # 2D array of articles
+    return [] unless articles.any?
 
-    row = 0
-    col = 0
-    articles.each do |article|
+    rows = (articles.length.to_f / LIST_NUM_COLUMNS).ceil
+    vertical_articles = Array.new(rows) { Array.new(LIST_NUM_COLUMNS) }
+
+    articles.each_with_index do |article, index|
+      row = index % rows
+      col = index / rows
       vertical_articles[row][col] = article
-      row+=1          # Go down a row each time
-      if row == rows  # When you reach the bottom, move to the next column
-        col+= 1
-        row = 0
-      end
     end
 
-    return vertical_articles
+    vertical_articles
   end
 
-  private
+  def find_or_create_category(collection, title)
```

Comment:
Moved this to a `private` function. It doesn't seem to be a route. Ideally we move helper functions like these to `private` and keep all public functions as `route/actions` only to help with readability.